src/Components/AnimationCustomization/AnimationCustomization.tsx

```diff
@@ -0,0 +1,105 @@
+import "./AnimationCustomization.css";
+import { useEffect, useState } from "react";
+import { animationProperties, SelectedAnimation } from '../../Types/Figures';
+import gravity from '../../Images/gravity.png';
+import gravity2 from '../../Images/gravity-v2.png';
+import radial from '../../Images/radial.png';
+import radial2 from '../../Images/radial-v2.png';
+import bubbles from '../../Images/bubbles.png';
+import draggedOut from '../../Images/draggedout.png';
+import draggedPainting from '../../Images/draggedPainting.png';
+import freeDraw from '../../Images/free-draw.png';
+
+export default function AnimationCustomization(props) {
+
+    const [img0, setImg0] = useState();
+    const [img1, setImg1] = useState();
+    const [img2, setImg2] = useState();
+
+    let imgSetters = {
+        "animation0": setImg0,
+        "animation1": setImg1,
+        "animation2": setImg2,
+    }
+
+    useEffect(() => {
+        for (let animation in props.animations) {
+            if (props.animations[animation] !== SelectedAnimation.None) {
+                imgSetters[animation](
+                    animationProperties(props.animations[animation])["image"]
+                );
+            }
+            else {
+                imgSetters[animation](null);
+            }
+        }
+    }, [props.animations]);
+
+    function createAnimationRow(anims) {
+        return (
+            <div className="unused-animation-row">
+                {
+                    anims.map((a) => {
+                        let sa = SelectedAnimation[a as keyof typeof SelectedAnimation];
+                        let selected = Object.values(props.animations).includes(sa) ? "animation-selected" : "";
+                        return <div className={"unused-animation " + selected} onClick={() => props.animationAddHandler(sa)}>
+                                <img className="animation-icon" src={animationProperties(sa)["image"]} alt={animationProperties(sa)["name"]} />
+                        </div>
+                    })
+                }
+            </div>
+        )
+    }
+
+    let animationDivs = [];
+    let animationRows = [];
+    let allAnimations = [
+        SelectedAnimation.WobblySwarm,
+        SelectedAnimation.DownwardGravity,
+        SelectedAnimation.DraggedOut,
+        SelectedAnimation.BubblePop,
+        SelectedAnimation.RadialForce,
+        SelectedAnimation.DraggedPainting,
+        // SelectedAnimation.WallBounce,
+        // SelectedAnimation.Stack,
+    ]
+    for (let a in allAnimations) {
+        let sa: SelectedAnimation = SelectedAnimation[a as keyof typeof SelectedAnimation]; //the stupidest type conversion i have ever seen
```

Comment:
If type conversions and constructing keys to access dictionaries is too much of a hassle I think you could make animationProperties into an array so that you can easily loop through all of the animations as well as index into it using the SelectedAnimation enum. 

Not really an issue though, its just a thought.